libraries/AP_ADSB/AP_ADSB_Sagetech.h

```diff
@@ -0,0 +1,157 @@
+class AP_ADSB_Sagetech : public AP_ADSB_Backend {
+public:
+    using AP_ADSB_Backend::AP_ADSB_Backend;
+
+    // init - performs any required initialisation for this instance
+    bool init() override;
+
+    // update - should be called periodically
+    void update() override;
+
+    // static detection function
+    static bool detect();
+
+private:
+
+    static const uint8_t PAYLOAD_XP_MAX_SIZE  = 52;
+    //static const uint8_t PAYLOAD_MX_MAX_SIZE  = 250;
+
+    enum class SystemStateBits {
+        Error_Transponder       = (1U<<0),
+        Altitidue_Source        = (1U<<1),
+        Error_GPS               = (1U<<2),
+        Error_ICAO              = (1U<<3),
+        Error_Over_Temperature  = (1U<<4),
+        Error_Extended_Squitter = (1U<<5),
+        Mode_Transponder        = (3U<<6),   // 2 bit status:
+    };
+
+    enum class Transponder_Type {
+        Mode_C                  = 0x00,
+        Mode_S_ADSB_OUT         = 0x01,
+        Mode_S_ADSB_OUT_and_IN  = 0x02,
+        Unknown                 = 0xFF,
+    };
+
+    enum class MsgTypes_XP {
+        INVALID                 = 0,
+        Installation_Set        = 0x01,
+        Preflight_Set           = 0x02,
+        Operating_Set           = 0x03,
+        GPS_Set                 = 0x04,
+        Request                 = 0x05,
+
+        ACK                     = 0x80,
+        Installatioin_Response  = 0x81,
+        Preflight_Response      = 0x82,
+        Status_Response         = 0x83,
+        ADSB_StateVector_Report = 0x91,
+        ADSB_ModeStatus_Report  = 0x92,
+        TISB_StateVector_Report = 0x93,
+        TISB_ModeStatus_Report  = 0x94,
+        TISB_CorasePos_Report   = 0x95,
+        TISB_ADSB_Mgr_Report    = 0x96,
+    };
+
+    enum class Protocol {
+        UNKNOWN,
+        XP,
+        MX,
+    } protocol;
+
+    enum class ParseState {
+        WaitingFor_Start,
+        WaitingFor_AssmAddr,
+        WaitingFor_MsgType,
+        WaitingFor_MsgId,
+        WaitingFor_PayloadLen,
+        WaitingFor_PayloadContents,
+        WaitingFor_ChecksumFletcher,
+        WaitingFor_Checksum,
+        WaitingFor_End,
+    };
+
+    struct Packet_XP {
+        const uint8_t   start = 0xA5;
+        const uint8_t   assemAddr = 0x01;
+        MsgTypes_XP  type;
+        uint8_t         id;
+        uint8_t         payload_length;
+        uint8_t         payload[PAYLOAD_XP_MAX_SIZE];
+        uint8_t         checksumFletcher;
+        uint8_t         checksum;
+        const uint8_t   end = 0x5A;
+    };
+
+    struct {
+        ParseState      state;
+        uint8_t         index;
+        Packet_XP       packet;
+    } message_in_xp;
+
+    // protocol XP
+    bool checksum_XP(Packet_XP &msg);
+    bool parse_byte_XP(const uint8_t data);
+    void parse_packet_XP(const Packet_XP &msg);
+    void send_msg(Packet_XP &msg);
+
+    // protocol MX
+    bool parse_byte_MX(const uint8_t data) { return false; }
+    void parse_packet_MX() {}

```

Comment:
If we're not going to handle MX as part of this, please remove all vestiges of it.

You should be able to do so while keeping those patches with sufficiently deft use of `git revert -n`